modules/store/spec/integration_signals.spec.ts

```diff
@@ -130,4 +131,30 @@ describe('NgRx and Signals Integration spec', () => {
       expect(error).toBeUndefined();
     });
   });
+
+  describe('immutable state integration spec', () => {
+    it('Store.selectSignal should not trigger on unrelated global state changes', () => {
+      let todosTriggerCount = 0;
+
+      const todos = TestBed.runInInjectionContext(() =>
+        store.selectSignal((state) => state.todos)
+      );
+
+      const todosTriggerState = TestBed.runInInjectionContext(() =>
+        computed(() => {
+          todos();
+          return ++todosTriggerCount;
+        })
+      );
```

Comment:
we should be able to use `selectSignal` and `computed` out of injection context:

```suggestion
      const todos = store.selectSignal((state) => state.todos);

      const todosTriggerState = computed(() => {
        todos();
        return ++todosTriggerCount;
      });
```